Sync without addr causes panic

Issuing the sync command without an address in the command is causing our indexer to panic and crash

We ran the following:

`storetheindex@net-manage01:~/.storetheindex$ storetheindex admin sync -p 12D3KooWAy5kaLtHf5uS7PZVLjSYd8sGqJ6fn7bxMjqLLZ1uULp9
Syncing request accepted. Come back later to check if syncing was successful`

and got:

```{"level":"info","ts":"2023-11-13T09:36:28.663Z","logger":"indexer/admin","caller":"admin/handler.go:331","msg":"Syncing with peer","peerID":"12D3KooWAy5kaLtHf5uS7PZVLjSYd8sGqJ6fn7bxMjqLLZ1uULp9"}
{"level":"info","ts":"2023-11-13T09:36:28.663Z","logger":"indexer/ingest","caller":"ingest/ingest.go:417","msg":"Explicitly syncing the latest advertisement","peer":"12D3KooWAy5kaLtHf5uS7PZVLjSYd8sGqJ6fn7bxMjqLLZ1uULp9","addrs":[null],"depth":0,"resync":false}
{"level":"warn","ts":"2023-11-13T09:36:28.663Z","logger":"peerstore","caller":"pstoremem/addr_book.go:244","msg":"Was passed nil multiaddr","peer":"12D3KooWAy5kaLtHf5uS7PZVLjSYd8sGqJ6fn7bxMjqLLZ1uULp9"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x7a5eed]

goroutine 6549 [running]:
github.com/multiformats/go-multiaddr.SplitFunc({0x0?, 0x0?}, 0xc003ffb328)
        /home/storetheindex/go/pkg/mod/github.com/multiformats/go-multiaddr@v0.12.0/util.go:129 +0x10d
github.com/libp2p/go-libp2p/p2p/http.normalizeHTTPMultiaddr({0x0, 0x0})
        /home/storetheindex/go/pkg/mod/github.com/libp2p/go-libp2p@v0.32.0/p2p/http/libp2phttp.go:710 +0x65
github.com/libp2p/go-libp2p/p2p/http.(*Host).NewConstrainedRoundTripper(0xc005522000, {{0xc00162e060, 0x26}, {0xc005a60000, 0x1, 0x1}}, {0x0, 0x0, 0xc000580800?})
        /home/storetheindex/go/pkg/mod/github.com/libp2p/go-libp2p@v0.32.0/p2p/http/libp2phttp.go:609 +0x1c7
github.com/libp2p/go-libp2p/p2p/http.(*Host).NamespacedClient(0x1bf76bf?, {0x1a9ffbb, 0xb}, {{0xc00162e060, 0x26}, {0xc005a60000, 0x1, 0x1}}, {0x0, 0x0, ...})
        /home/storetheindex/go/pkg/mod/github.com/libp2p/go-libp2p@v0.32.0/p2p/http/libp2phttp.go:570 +0xc5
github.com/ipni/go-libipni/dagsync/ipnisync.(*Sync).NewSyncer(0xc005500000, {{0xc00162e060, 0x26}, {0xc005a60000, 0x1, 0x1}})
        /home/storetheindex/go/pkg/mod/github.com/ipni/go-libipni@v0.5.5/dagsync/ipnisync/sync.go:103 +0x190
github.com/ipni/go-libipni/dagsync.(*Subscriber).makeSyncer(0xc005558000, {{0xc00162e060, 0x26}, {0xc005a60000, 0x1, 0x1}}, 0x1)
        /home/storetheindex/go/pkg/mod/github.com/ipni/go-libipni@v0.5.5/dagsync/subscriber.go:824 +0x2f0
github.com/ipni/go-libipni/dagsync.(*Subscriber).SyncAdChain(0xc005558000, {0x1f2b330, 0xc003e023c0}, {{0xc00162e060, 0x26}, {0xc005a60000, 0x1, 0x1}}, {0x0, 0x0, ...})
        /home/storetheindex/go/pkg/mod/github.com/ipni/go-libipni@v0.5.5/dagsync/subscriber.go:424 +0x43d
github.com/ipni/storetheindex/internal/ingest.(*Ingester).Sync(0xc002986b60, {0x1f2b2c0, 0xc0104634f0}, {{0xc00162e060, 0x26}, {0xc005a60000, 0x1, 0x1}}, 0x0, 0x0)
        /home/storetheindex/go/pkg/mod/github.com/ipni/storetheindex@v0.8.7/internal/ingest/ingest.go:462 +0x6de
github.com/ipni/storetheindex/server/admin.(*adminHandler).handlePostSyncs.func1()
        /home/storetheindex/go/pkg/mod/github.com/ipni/storetheindex@v0.8.7/server/admin/handler.go:338 +0x13c
created by github.com/ipni/storetheindex/server/admin.(*adminHandler).handlePostSyncs
        /home/storetheindex/go/pkg/mod/github.com/ipni/storetheindex@v0.8.7/server/admin/handler.go:333 +0xe98
```

--- a/go.mod
+++ b/go.mod
@@ -23,8 +23,8 @@ require (
 	github.com/ipld/go-ipld-prime v0.21.0
 	github.com/ipld/go-ipld-prime/storage/dsadapter v0.0.0-20230102063945-1a409dc236dd
 	github.com/ipni/go-indexer-core v0.8.2
-	github.com/ipni/go-libipni v0.5.5
-	github.com/libp2p/go-libp2p v0.32.0
+	github.com/ipni/go-libipni v0.5.6
+	github.com/libp2p/go-libp2p v0.32.1
 	github.com/libp2p/go-msgio v0.3.0
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/multiformats/go-multiaddr v0.12.0

--- a/server/admin/handler.go
+++ b/server/admin/handler.go
@@ -332,9 +332,12 @@ func (h *adminHandler) handlePostSyncs(w http.ResponseWriter, r *http.Request) {
 	h.pendingSyncs.Add(1)
 	go func() {
 		peerInfo := peer.AddrInfo{
-			ID:    peerID,
-			Addrs: []multiaddr.Multiaddr{syncAddr},
+			ID: peerID,
 		}
+		if syncAddr != nil {
+			peerInfo.Addrs = []multiaddr.Multiaddr{syncAddr}
+		}
+
 		_, err := h.ingester.Sync(h.ctx, peerInfo, int(depth), resync)
 		if err != nil {
 			log.Errorw("Cannot sync with peer", "err", err)

--- a/version.json
+++ b/version.json
@@ -1,3 +1,3 @@
 {
-  "version": "v0.8.7"
+  "version": "v0.8.8"
 }

